source/dem/particle_wall_contact_force.cc

```diff
@@ -158,209 +165,445 @@ ParticleWallContactForce<dim,
   const double force_calculation_threshold_distance =
     get_force_calculation_threshold_distance();
 
+  // Initiate containers
   std::vector<Particles::ParticleIterator<dim>> particle_locations;
   std::vector<Point<dim>> triangle(this->vertices_per_triangle);
 
   // Iterating over the solid objects
   for (unsigned int solid_counter = 0; solid_counter < solids.size();
        ++solid_counter)
     {
-      // Get translational and rotational velocities and center of
-      // rotation of solid object
-      Tensor<1, 3> translational_velocity =
-        solids[solid_counter]->get_translational_velocity();
-      Tensor<1, 3> angular_velocity =
-        solids[solid_counter]->get_angular_velocity();
-      Point<3> center_of_rotation =
-        solids[solid_counter]->get_center_of_rotation();
-      Parameters::ThermalBoundaryType thermal_boundary_type =
-        solids[solid_counter]->get_thermal_boundary_type();
-      double temperature_wall;
-      if (thermal_boundary_type != Parameters::ThermalBoundaryType::adiabatic)
-        temperature_wall = solids[solid_counter]->get_temperature();
-
-      auto &particle_floating_mesh_contact_pair =
-        particle_floating_mesh_in_contact[solid_counter];
-
-      for (auto &[cut_cell, map_info] : particle_floating_mesh_contact_pair)
+      // Extract the neighboring cell maps of this solid objects
+      const auto &[this_solid_es_neighbors, this_solid_vs_neighbors] =
+        solids[solid_counter]->get_neighbors_maps();
+
+      // For each solid surface, we create a map used to store every contact.
+      // The key of that map is the particle local ID.
+      // The value is a vector of tuple storing the required information to
+      // compute the contact force later on.
+      // The information includes: 1. The triangle cell with which the contact
+      // is occurring, 2. The normal overlap, 3. The type of contact (face, edge
+      // or vertex) 4. The contact info associated.
+      using particle_triangle_contact_description = std::vector<
+        std::tuple<typename Triangulation<dim - 1, dim>::active_cell_iterator,
+                   double,
+                   LetheGridTools::ParticleTriangleContactIndicator,
+                   particle_wall_contact_info<dim>>>;
+
+      using particle_triangle_contact_record =
+        ankerl::unordered_dense::map<types::particle_index,
+                                     particle_triangle_contact_description>;
```

Comment:
Don't you want instead to use typedefs in the .h of that file? It will be a lot clearer to use a typedef instead of a using locally I find. I'm a bit skeptical of these using statements in only a single routine. These are quite big structures, In that case it's better to have a typedef generally.